test/pkg/firewall/firewall.go

```diff
@@ -0,0 +1,140 @@
+	if err != nil {
+		return fmt.Errorf("failed to save rule set to file on node %s: %w", nodeName, err)
+	}
+
+	// run apply nft command for the rules
+	_, err = RunCommandInPod(debugPod, "nft -f /etc/nftables/firewall.nft", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to apply rule set on node %s: %w", nodeName, err)
+	}
+
+	output, err := nftList(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	// edit the sys conf to make sure it will be on the list after reboot
+	err = editNftablesConf(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	_, err = RunCommandInPod(debugPod, "/usr/sbin/nft list ruleset > /etc/nftables.conf", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", nodeName, err)
+	}
+
+	_, err = RunCommandInPod(debugPod, "systemctl enable nftables", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to enable nftables on node %s: %w", nodeName, err)
+	}
+
+	err = utilsHelpers.WriteFile(filepath.Join(artifactsDir, "nftables-"+nodeName), output)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func RulesList(nodeName, artifactsDir string, utilsHelpers utils.UtilsInterface) ([]byte, error) {
+	debugPod, err := utilsHelpers.CreatePodOnNode(nodeName, consts.DefaultDebugNamespace, consts.DefaultDebugPodImage)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create debug pod on node %s: %w", nodeName, err)
+	}
+
+	defer func() {
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			log.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	output, err := nftList(debugPod, utilsHelpers)
+	if err != nil {
+		return nil, err
+	}
+
+	err = utilsHelpers.WriteFile(filepath.Join(artifactsDir, "nftables-after-reboot"+nodeName), output)
+	if err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+func RunCommandInPod(debugPod *v1.Pod, command string, chrootDir bool, utilsHelpers utils.UtilsInterface) ([]byte, error) {
+	if chrootDir {
+		// Format the command for chroot
+		command = fmt.Sprintf("chroot /host /bin/bash -c '%s'", command)
+	}
+
+	output, err := utilsHelpers.RunCommandOnPod(debugPod, []string{"bash", "-c", command})
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute command '%s' on node %s: %w", command, debugPod.Spec.NodeName, err)
+	}
+	return output, nil
+}
+
+// Checks if the include statement for the new nftables configuration,
+// Is present in the nftables.conf file. If the include statement is not present,
+// Add it the statement to the file.
+func editNftablesConf(debugPod *v1.Pod, utilsHelpers utils.UtilsInterface) error {
+	checkCommand := `grep -q 'include "/etc/nftables/firewall.nft"' /host/etc/sysconfig/nftables.conf`
+	_, err := RunCommandInPod(debugPod, checkCommand, false, utilsHelpers)
+	if err == nil {
```

Comment:
This is not right. the `RunCommandInPod` command can failed because of other issues like connection to pod, command run issue etc..

you can have a case where running the exec failed (but the file exist) and you will add it again.

you can do something like `grep 'include "/etc/nftables/firewall.nft"' /host/etc/sysconfig/nftables.conf | wc -l` and check the output is 0 or 1